hal/src/nRF52840/openthread/radio.c

```diff
@@ -278,30 +300,18 @@ otError otPlatRadioEnable(otInstance *aInstance)
 
 otError otPlatRadioDisable(otInstance *aInstance)
 {
-    otError error;
+    otError error = OT_ERROR_NONE;
 
-    OT_UNUSED_VARIABLE(aInstance);
+    otEXPECT(otPlatRadioIsEnabled(aInstance));
+    otEXPECT_ACTION(otPlatRadioGetState(aInstance) == OT_RADIO_STATE_SLEEP || isPendingEventSet(kPendingEventSleep),
+                    error = OT_ERROR_INVALID_STATE);
 
-    if (!sDisabled)
-    {
-        sDisabled = true;
-        error     = OT_ERROR_NONE;
-    }
-    else
-    {
-        error = OT_ERROR_INVALID_STATE;
-    }
+    sDisabled = true;
 
+exit:
     return error;
 }
 
-bool otPlatRadioIsEnabled(otInstance *aInstance)
```

Comment:
These sources come (modified) from OpenThread [`examples`](https://github.com/particle-iot/openthread/tree/e2d61d7adc142cf5fba096f9831d3801d1d93ef6/examples/platforms/nrf52840) folder. I'd rather not modify the original example sources within our OpenThread fork and unless we create a separate repository just so that we could have them in `third_party` folder, I'd prefer to keep things as-is. We have other cases like this as well: e.g. `hal/src/nRF52840/mbedtls/`. I'm open to other suggestions to improve this.